### PR TITLE
Implement provisioning independent thin VM

### PR DESCRIPTION
**Pull request by borod108**
Add the show the linked_clone button in the provisioning dialog so the
user can specify if they want the clone to be linked(dependant) or not
(clone/independent)

requires: https://github.com/ManageIQ/manageiq/pull/18833
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1712934

### CODE REVIEW

File: app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb [hunk 1]
```diff
@@ -150,13 +150,13 @@ def update_field_visibility
     super(:force_platform => 'linux')
   end
 
-  def update_field_visibility_linked_clone(_options = {}, f)
-    show_flag = supports_native_clone? ? :edit : :hide
-    f[show_flag] << :linked_clone
+  #def update_field_visibility_linked_clone(_options = {}, f)
```
Comment (doron-fediuck) [inline app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb]: If this is no longer needed, why not remove it all together?

**Comment by borod108** [comment]
@agrare what do you think?

**Comment by borod108** [comment]
Also the rubocop is wrong in my opinion in this case, I have a comment there instead of an empty line, it looks better that way in my opinion.

**Comment by borod108** [comment]
@masayag can you please review?

File: app/models/manageiq/providers/redhat/infra_manager/provision/cloning.rb [hunk 2]
```diff
@@ -36,15 +36,33 @@ def prepare_clone_options
...
     }
     clone_options[:storage] = dest_datastore.ems_ref unless dest_datastore.nil?
     clone_options
   end
 
+  def disk_format
+    # If the datastore is not set, we will go with "block", because this will ensure
+    # the result of the sparsity will be as set by the user.
+    ds_type = "block" unless dest_datastore
+    ds_type ||= dest_datastore.store_type.to_s.upcase == 'iscsi'.upcase ? 'block' : 'file'
```
Comment (masayag) [inline app/models/manageiq/providers/redhat/infra_manager/provision/cloning.rb]: you can replace`dest_datastore.store_type.to_s.upcase == 'iscsi'.upcase` with
dest_datastore.store_type.to_s.downcase == 'iscsi'

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 2b9a0db19c16d9e538e71dd6fdf6e038ad34382a
Author: Moti Asayag <masayag@redhat.com>
Date:   Thu Jun 13 15:17:03 2019 +0300

    Merge pull request #372 from borod108/bugs/1712934linked_clone
    
    Implement provisioning independent thin VM
    
    (cherry picked from commit ec246d326fa92b095271e7a2925af7e66ca9a039)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1720320
```

**Pull request merged**